app/lib/search/index_simple.dart

```diff
@@ -168,6 +161,80 @@ class SimplePackageIndex implements PackageIndex {
       value: (String url) => _documents[url].popularity * 100,
     );
   }
+
+  Score _searchText(String text, String packagePrefix) {
+    Score textScore;
+    if (text != null && text.isNotEmpty) {
+      textScore = new Score()
+        ..addValues(_nameIndex.search(text), 0.82)
+        ..addValues(_descrIndex.search(text), 0.12)
+        ..addValues(_readmeIndex.search(text), 0.06);
+      textScore.removeLowScores(0.05);
+    } else if (packagePrefix != null) {
+      textScore = new Score()..addValues(_nameIndex.search(packagePrefix), 0.8);
+    }
+    return textScore;
+  }
+
+  List<PackageScore> _flattenFromValues(Map<String, double> values) {
+    final List<PackageScore> list = values.keys
+        .map((url) => new PackageScore(
+              url: url,
+              package: _documents[url].package,
+              score: values[url],
+            ))
+        .toList();
+    list.sort((a, b) {
+      final int scoreCompare = -a.score.compareTo(b.score);
+      if (scoreCompare != 0) return scoreCompare;
+      // if two packages got the same score, order by last updated
+      return _compareUpdated(_documents[a.url], _documents[b.url]);
+    });
+    return list;
+  }
+
+  List<PackageScore> _flattenFromUrls(
+      Set<String> urls, int compare(PackageDocument a, PackageDocument b)) {
+    final List<PackageScore> list = urls
+        .map((url) =>
+            new PackageScore(url: url, package: _documents[url].package))
```

Comment:
It's not clear why you are not adding a `, score: values[url]` here!

Reply:
This is the use case for e.g. sort by `updated`, where we don't have scores (no similarity, no health/popularity).